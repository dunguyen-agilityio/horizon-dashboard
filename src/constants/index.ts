export * from './common';
export * from './images';
export * from './messages';
export * from './routes';
export * from './text';
export * from './endpoint';
export * from './session';
