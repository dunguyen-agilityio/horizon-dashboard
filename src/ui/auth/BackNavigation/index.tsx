'use client';

// Libs
import Link from 'next/link';
import { usePathname } from 'next/navigation';

// Components
import { ArrowRight } from '@/icons';
import { BoxIcon, Text } from '@/components';

// Constants
import {
  AUTH_NAVIGATION_HEADER,
  AUTH_ROUTES,
  PUBLIC_ROUTES,
} from '@/constants';

// Types
import { TEXT_VARIANT } from '@/types/text';

const BackNavigation = () => {
  const pathname = usePathname();

  const getNavigationHeader = (): { title: string; href: string } => {
    switch (true) {
      case pathname.includes(AUTH_ROUTES.SIGN_IN):
        return {
          title: AUTH_NAVIGATION_HEADER[PUBLIC_ROUTES.DASHBOARD],
          href: PUBLIC_ROUTES.DASHBOARD,
        };

      case pathname.includes(AUTH_ROUTES.FORGET_PASSWORD):
      case pathname.includes(AUTH_ROUTES.SIGN_UP):
      case pathname.includes(AUTH_ROUTES.RESET_PASSWORD):
        return {
          title: AUTH_NAVIGATION_HEADER[AUTH_ROUTES.SIGN_IN],
          href: AUTH_ROUTES.SIGN_IN,
        };

      default:
        return { title: '', href: '' };
    }
  };

  const { title, href } = getNavigationHeader();

  return (
    <Link href={href} className="h-fit flex items-center gap-1">
      <BoxIcon icon={<ArrowRight />} customClass="fill-secondary" />
      <Text variant={TEXT_VARIANT.SECONDARY} className="ml-1">
        {`Back to ${title}`}
      </Text>
    </Link>
  );
};

export default BackNavigation;
