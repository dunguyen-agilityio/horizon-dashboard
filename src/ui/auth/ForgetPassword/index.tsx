'use client';

import { useState } from 'react';

// Components
import ConfirmEmailForm from './ConfirmEmailForm';
import ResendEmailConfirm from './ResendEmailConfirm';

const ForgetPasswordForm = () => {
  const [email, setEmail] = useState('');

  const handleSendEmailConfirm = async (email: string) => {
    setEmail(email);
  };

  const renderContent = () => {
    switch (true) {
      case !email:
        return <ConfirmEmailForm onSendEmailConfirm={handleSendEmailConfirm} />;
      default:
        return <ResendEmailConfirm email={email} />;
    }
  };

  return <div className="w-full sm:w-96">{renderContent()}</div>;
};

export default ForgetPasswordForm;
