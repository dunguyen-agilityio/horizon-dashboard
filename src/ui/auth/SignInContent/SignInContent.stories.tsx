import type { Meta, StoryObj } from '@storybook/react';

import SignInForm from '@/ui/auth/SignInContent';

const meta = {
  title: 'Components/SignInForm',
  component: SignInForm,
  tags: ['autodocs'],
} satisfies Meta<typeof SignInForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
