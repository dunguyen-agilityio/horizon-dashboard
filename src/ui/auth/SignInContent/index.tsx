import { Controller, useForm } from 'react-hook-form';

// Components
import { Link } from '@nextui-org/link';
import { Input } from '@nextui-org/input';
import { Checkbox } from '@nextui-org/checkbox';
import { Button, InputPassword, Text } from '@/components';

// Types
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';
import type { UserSignin } from '@/types/authentication';

// Utils
import { isEnableSubmitButton } from '@/utils/validation';

// Schema
import { useMemo } from 'react';

const REQUIRED_FIELDS = ['email', 'password'];

const SignInContent = () => {
  const signinFormInitValues: UserSignin = {
    email: '',
    password: '',
  };

  const {
    register,
    control,
    formState: { dirtyFields, errors },
  } = useForm<UserSignin>({
    mode: 'onBlur',
    reValidateMode: 'onBlur',
    values: signinFormInitValues,
  });

  const dirtyItems = Object.keys(dirtyFields);
  const isDisabled = useMemo(
    () => !isEnableSubmitButton(REQUIRED_FIELDS, dirtyItems, errors),
    [dirtyItems, errors],
  );

  return (
    <>
      <div className="pt-3 mb-8 flex flex-col gap-6">
        <Controller
          control={control}
          {...register('email', { required: 'Email is required' })}
          render={({
            field: { onChange, value, ...rest },
            fieldState: { error },
          }) => (
            <Input
              isRequired
              size="lg"
              key="email"
              type="email"
              label="Email"
              labelPlacement="outside"
              placeholder="mail@simmmple.com"
              value={value}
              onChange={(value) => {
                onChange(value);
              }}
              isInvalid={!!errors.email}
              errorMessage={error?.message}
              {...rest}
            />
          )}
        />
        <Controller
          control={control}
          {...register('password', { required: 'Password is required' })}
          render={({
            field: { onChange, value, ...rest },
            fieldState: { error },
          }) => (
            <InputPassword
              value={value}
              onChange={(value) => {
                onChange(value);
              }}
              isInvalid={!!errors.password}
              errorMessage={error?.message}
              {...rest}
            />
          )}
        />
      </div>

      <div className="flex justify-between">
        <Checkbox
          aria-label="checkbox-keep-login"
          defaultSelected
          radius="sm"
          className="fill-blue-450"
        >
          <Text as="p" size={TEXT_SIZE['md']}>
            Keep me logged in
          </Text>
        </Checkbox>
        <Link href="/forget-password">
          <Text
            size={TEXT_SIZE.sm}
            variant={TEXT_VARIANT.QUATERNARY}
            className="pr-3"
          >
            Forget password?
          </Text>
        </Link>
      </div>

      <Button
        isDisabled={isDisabled}
        className="bg-blue-450 dark:bg-purple-750 w-full py-7 mb-6 mt-8"
      >
        <Text size={TEXT_SIZE.sm} className="text-white font-bold leading-4">
          Sign In
        </Text>
      </Button>
    </>
  );
};

export default SignInContent;
