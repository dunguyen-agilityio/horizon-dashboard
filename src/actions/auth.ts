'use server';

// Libs
import { cookies } from 'next/headers';

// Config
import { signIn, signOut } from '@/auth.config';

// Constants
import { API_ENTITY } from '@/constants/api';
import { API_TOKEN, AUTH_ROUTES } from '@/constants';

// Types
import { ForgetResponse, SignInFormData, SignInResponse } from '@/types/auth';

// Services
import { apiClient } from '@/services/api';

/**
 *
 * @param formData SignInFormData
 * @returns hasError boolean
 */
export const handleSignIn = async (formData: SignInFormData) =>
  await signIn('credentials', { ...formData, redirect: false });

export const handleSignOut = async () => {
  await signOut({ redirectTo: AUTH_ROUTES.SIGN_IN });
  cookies().delete(API_TOKEN);
};

export const handleForgetPassword = async (email: string) =>
  await apiClient.post<ForgetResponse>(API_ENTITY.FORGET_PASSWORD, {
    body: { email },
  });

export type ResetPasswordPayload = {
  code: string;
  password: string;
  passwordConfirmation: string;
};

export const handleResetPassword = async (payload: ResetPasswordPayload) =>
  await apiClient.post<SignInResponse>(API_ENTITY.RESET_PASSWORD, {
    body: payload,
  });
