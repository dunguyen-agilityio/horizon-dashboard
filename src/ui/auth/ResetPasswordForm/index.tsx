'use client';

// Libs
import { useRouter } from 'next/navigation';
import { Controller, useForm } from 'react-hook-form';

// Components
import { Button, InputPassword, Text } from '@/components';

// Types
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';

// Constants
import { AUTH_ROUTES, MESSAGES, REGEX_PASSWORD } from '@/constants';

// Actions
import { handleResetPassword } from '@/actions/auth';

// Hooks
import useToast from '@/contexts/toast';

interface ResetPasswordFormData {
  password: string;
  passwordConfirmation: string;
}

const resetPasswordInitValues: ResetPasswordFormData = {
  password: '',
  passwordConfirmation: '',
};

interface ResetPasswordFormProps {
  code: string;
}

const { PASSWORD, RESET_PASSWORD } = MESSAGES;

const ResetPasswordForm = ({ code }: ResetPasswordFormProps) => {
  const {
    control,
    getValues,
    handleSubmit,
    formState: { isDirty },
  } = useForm<ResetPasswordFormData>({
    mode: 'onBlur',
    values: resetPasswordInitValues,
  });

  const { showToast } = useToast();
  const { push } = useRouter();

  const isDisabled = !isDirty;

  const resetPassword = async (data: ResetPasswordFormData) => {
    const { data: res } = await handleResetPassword({ code, ...data });
    const { ERROR, SUCCESS } = RESET_PASSWORD;

    const isSuccess = !!res?.user;

    const { TITLE, DESCRIPTION } = isSuccess ? SUCCESS : ERROR;

    showToast({
      title: TITLE,
      message: DESCRIPTION,
      type: isSuccess ? 'success' : 'error',
    });

    isSuccess && push(AUTH_ROUTES.SIGN_IN);
  };

  return (
    <form onSubmit={handleSubmit(resetPassword)} className="w-full sm:w-96">
      <div className="flex flex-col gap-6">
        <div className="flex flex-col gap-2 pb-8">
          <Text as="h1" size={TEXT_SIZE['2xl']} className="leading-[56px]">
            Reset your password
          </Text>
          <Text
            as="h2"
            size={TEXT_SIZE['md']}
            variant={TEXT_VARIANT.TERTIARY}
            className="leading-6"
          >
            Enter your new password to reset your password.
          </Text>
        </div>
        <Controller
          name="password"
          control={control}
          rules={{
            required: PASSWORD.REQUIRED,
            pattern: {
              value: REGEX_PASSWORD,
              message: PASSWORD.REGEX_MISMATCH,
            },
          }}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              isInvalid={invalid}
              errorMessage={error?.message}
              label="New Password"
              placeholder="New Password"
              {...field}
            />
          )}
        />
        <Controller
          name="passwordConfirmation"
          control={control}
          rules={{
            required: PASSWORD.REQUIRED,
            validate: (value) =>
              value === getValues('password') ? true : PASSWORD.MISMATCH,
          }}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              isInvalid={invalid}
              errorMessage={error?.message}
              label="Confirm Password"
              placeholder="Confirm Password"
              {...field}
            />
          )}
        />
        <Button
          isDisabled={isDisabled}
          className="w-full py-7 mt-6"
          type="submit"
          data-testid="update-password-btn"
        >
          <Text size={TEXT_SIZE.sm} className="text-white font-bold leading-4">
            Update Password
          </Text>
        </Button>
      </div>
    </form>
  );
};

export default ResetPasswordForm;
