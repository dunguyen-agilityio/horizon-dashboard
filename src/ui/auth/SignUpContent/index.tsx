'use client';

import { useState } from 'react';
import { useRouter } from 'next/navigation';
import { Controller, useForm } from 'react-hook-form';

// Components
import { Input } from '@nextui-org/input';
import { Button, InputPassword, Text } from '@/components';

// Types
import { TEXT_SIZE } from '@/types/text';
import type { SignUpFormData, SignUpPayload } from '@/types/auth';

// Regex
import { REGEX_EMAIL, REGEX_PASSWORD } from '@/constants/regex';

// Constants
import { AUTH_ROUTES } from '@/constants';

// Utils
import {
  emailRules,
  identifierRules,
  passwordRules,
  validateConfirmPassword,
} from '@/utils/validation';

// Actions
import { handleSignUp } from '@/actions/auth';

// Hooks
import useToast from '@/contexts/toast';

// Style
import { STYLE_INPUT } from '@/styles/input';

const signUpFormInitValues: SignUpFormData = {
  username: '',
  email: '',
  password: '',
  confirmPassword: '',
};

const SignUpContent = () => {
  const {
    watch,
    control,
    formState: { isDirty },
    handleSubmit,
  } = useForm<SignUpFormData>({
    mode: 'onBlur',
    values: signUpFormInitValues,
  });

  const [isLoading, setIsLoading] = useState(false);
  const isDisabled = !isDirty;
  const passwordValue = watch('password');

  const { replace } = useRouter();
  const { showToast } = useToast();

  const signUp = async ({ email, password, username }: SignUpPayload) => {
    setIsLoading(true);
    const { error } = await handleSignUp({ email, password, username });

    if (error) {
      showToast({
        title: 'Failed',
        message: error.message,
        type: 'error',
        timeOut: 5000,
      });
      setIsLoading(false);
      return;
    }

    replace(AUTH_ROUTES.SIGN_IN);
  };

  return (
    <form onSubmit={handleSubmit(signUp)} className="relative">
      <div className="pt-3 flex flex-col gap-6">
        <Controller
          name="email"
          control={control}
          rules={emailRules(REGEX_EMAIL)}
          render={({ field, fieldState: { error, invalid } }) => (
            <Input
              isRequired
              size="lg"
              label="Email"
              labelPlacement="outside"
              placeholder="Enter your email"
              isInvalid={invalid}
              classNames={STYLE_INPUT}
              errorMessage={error?.message}
              {...field}
            />
          )}
        />
        <Controller
          name="username"
          control={control}
          rules={identifierRules}
          render={({ field, fieldState: { error, invalid } }) => (
            <Input
              isRequired
              size="lg"
              label="User name"
              labelPlacement="outside"
              classNames={STYLE_INPUT}
              placeholder="Enter your user name"
              isInvalid={invalid}
              errorMessage={error?.message}
              {...field}
            />
          )}
        />
        <Controller
          name="password"
          control={control}
          rules={passwordRules(REGEX_PASSWORD)}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              isInvalid={invalid}
              classNames={STYLE_INPUT}
              errorMessage={error?.message}
              {...field}
            />
          )}
        />
        <Controller
          name="confirmPassword"
          control={control}
          rules={{
            required: 'This field is required',
            validate: (confirmPassword: string) =>
              validateConfirmPassword(passwordValue, confirmPassword),
          }}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              label="Confirm Password"
              isInvalid={invalid}
              classNames={STYLE_INPUT}
              errorMessage={error?.message}
              placeholder="Enter password confirm"
              {...field}
            />
          )}
        />
      </div>
      <Button
        isDisabled={isDisabled}
        isLoading={isLoading}
        className="bg-blue-450 dark:bg-purple-750 w-full py-7 mt-10"
        type="submit"
        data-testid="signup-btn"
      >
        {!isLoading && (
          <Text size={TEXT_SIZE.sm} className="text-white font-bold leading-4">
            Sign Up
          </Text>
        )}
      </Button>
      {isLoading && <div className="absolute inset-0" />}
    </form>
  );
};

export default SignUpContent;
