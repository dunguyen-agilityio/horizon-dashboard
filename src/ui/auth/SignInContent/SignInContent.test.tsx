import { act, fireEvent, render, waitFor } from '@/utils/test-utils';

// Components
import SignInContent from '@/ui/auth/SignInContent';

// Config
import { signIn } from '@/auth.config';

// Constants
import { PUBLIC_ROUTES } from '@/constants';

jest.mock('@/auth.config', () => ({
  signIn: jest.fn(),
}));

const mockReplace = jest.fn();

jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockImplementation(() => ({
    replace: mockReplace,
  })),
}));

describe('SignInContent tests', () => {
  it('Should match snapshot', () => {
    const { container } = render(<SignInContent />);
    expect(container).toMatchSnapshot();
  });

  it('Should render EyeFilled icon', () => {
    const { getByTestId } = render(<SignInContent />);
    expect(getByTestId('eye-filled-icon')).toBeInTheDocument();
  });

  it('Should render Email and Password label', () => {
    const { getByLabelText } = render(<SignInContent />);
    expect(getByLabelText('Email or username')).toBeInTheDocument();
    expect(getByLabelText('Password')).toBeInTheDocument();
  });

  it('Should render placeholder on input of Email and Password', () => {
    const { getByPlaceholderText } = render(<SignInContent />);
    expect(getByPlaceholderText('Your username or email')).toBeInTheDocument();
    expect(getByPlaceholderText('Min. 8 characters')).toBeInTheDocument();
  });

  it('Should render checkbox keep me logged in', () => {
    const { getByText } = render(<SignInContent />);
    expect(getByText('Keep me logged in')).toBeInTheDocument();
  });

  it('Should render EyeSlashFilled icon when click EyeFilled', () => {
    const { getByTestId } = render(<SignInContent />);
    const eyeFilledIcon = getByTestId('eye-filled-icon');

    fireEvent.click(eyeFilledIcon);
    expect(getByTestId('eye-slash-filled-icon')).toBeInTheDocument();
  });

  it('handleSignIn will triggered correctly when click Login Button', async () => {
    (signIn as jest.Mock).mockResolvedValue(true);
    const { getByLabelText, getByTestId } = render(<SignInContent />);

    const mockUserName = 'user123';
    const mockPassword = 'Abcd@1234';

    act(() => {
      fireEvent.change(getByLabelText('Email or username'), {
        target: { value: mockUserName },
      });

      const inputPassword = getByLabelText('inputPassword');
      fireEvent.change(inputPassword, { target: { value: mockPassword } });
    });

    fireEvent.submit(getByTestId('signin-btn'));

    await waitFor(() => {
      expect(signIn).toHaveBeenNthCalledWith(1, 'credentials', {
        password: mockPassword,
        identifier: mockUserName,
        redirect: false,
      });
    });

    expect(mockReplace).toHaveBeenNthCalledWith(1, PUBLIC_ROUTES.DASHBOARD);
  });
});
