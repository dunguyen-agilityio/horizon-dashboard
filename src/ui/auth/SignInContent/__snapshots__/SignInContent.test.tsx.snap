// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInContent tests Should match snapshot 1`] = `
<div>
  <div
    class="pt-3 mb-8 flex flex-col gap-6"
  >
    <div
      class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_12px)]"
      data-filled="true"
      data-filled-within="true"
      data-has-elements="true"
      data-has-label="true"
      data-required="true"
      data-slot="base"
    >
      <div
        class="h-full flex flex-col"
        data-slot="main-wrapper"
      >
        <div
          class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-12 min-h-12 rounded-large transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background"
          data-slot="input-wrapper"
          style="cursor: text;"
        >
          <label
            class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 after:content-['*'] after:text-danger after:ml-0.5 rtl:after:ml-[unset] rtl:after:mr-0.5 will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 text-medium group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_24px)] pe-2 max-w-full text-ellipsis overflow-hidden"
            data-slot="label"
            for="react-aria-:r0:"
            id="react-aria-:r1:"
          >
            Email
          </label>
          <div
            class="inline-flex w-full items-center h-full box-border"
            data-slot="inner-wrapper"
          >
            <input
              aria-label="Email"
              aria-labelledby="react-aria-:r0: react-aria-:r1:"
              aria-required="true"
              class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-medium group-data-[has-value=true]:text-default-foreground"
              data-slot="input"
              id="react-aria-:r0:"
              name="email"
              placeholder="mail@simmmple.com"
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_12px)]"
      data-filled="true"
      data-filled-within="true"
      data-has-elements="true"
      data-has-label="true"
      data-required="true"
      data-slot="base"
    >
      <div
        class="h-full flex flex-col"
        data-slot="main-wrapper"
      >
        <div
          class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-12 min-h-12 rounded-large transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background"
          data-slot="input-wrapper"
          style="cursor: text;"
        >
          <label
            class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 after:content-['*'] after:text-danger after:ml-0.5 rtl:after:ml-[unset] rtl:after:mr-0.5 will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 text-medium group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_24px)] pe-2 max-w-full text-ellipsis overflow-hidden"
            data-slot="label"
            for="react-aria-:r5:"
            id="react-aria-:r6:"
          >
            Password
          </label>
          <div
            class="inline-flex w-full items-center h-full box-border"
            data-slot="inner-wrapper"
          >
            <input
              aria-label="inputPassword"
              aria-labelledby="react-aria-:r5: react-aria-:r6:"
              aria-required="true"
              class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-medium group-data-[has-value=true]:text-default-foreground"
              data-has-end-content="true"
              data-slot="input"
              data-testid="input-password"
              id="react-aria-:r5:"
              name="password"
              placeholder="Min. 8 characters"
              type="password"
              value=""
            />
            <button
              aria-label="toggle password visibility"
              class="focus:outline-none"
              type="button"
            >
              <i
                class="fill-secondary w-6 h-6 flex justify-center items-center"
              >
                <svg
                  data-testid="eye-slash-filled-icon"
                  height="19"
                  width="22"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 4c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C9.74 4.13 10.35 4 11 4ZM1 1.27l2.28 2.28.46.46A11.804 11.804 0 0 0 0 9c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L18.73 19 20 17.73 2.27 0 1 1.27ZM6.53 6.8l1.55 1.55C8.03 8.56 8 8.78 8 9c0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2Zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01Z"
                  />
                </svg>
              </i>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex justify-between"
  >
    <label
      aria-label="checkbox-keep-login"
      class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 -m-2 fill-blue-450"
      data-selected="true"
    >
      <span
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          aria-label="checkbox-keep-login"
          aria-labelledby=":ra:"
          checked=""
          type="checkbox"
          value=""
        />
      </span>
      <span
        aria-hidden="true"
        class="relative inline-flex items-center justify-center flex-shrink-0 overflow-hidden before:content-[''] before:absolute before:inset-0 before:border-solid before:border-2 before:box-border before:border-default after:content-[''] after:absolute after:inset-0 after:scale-50 after:opacity-0 after:origin-center group-data-[selected=true]:after:scale-100 group-data-[selected=true]:after:opacity-100 group-data-[hover=true]:before:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background after:bg-primary after:text-primary-foreground text-primary-foreground w-5 h-5 mr-2 rtl:ml-2 rtl:mr-[unset] rounded-[calc(theme(borderRadius.medium)*0.5)] before:rounded-[calc(theme(borderRadius.medium)*0.5)] after:rounded-[calc(theme(borderRadius.medium)*0.5)] before:transition-colors group-data-[pressed=true]:scale-95 transition-transform after:transition-transform-opacity after:!ease-linear after:!duration-200 motion-reduce:transition-none"
      >
        <svg
          aria-hidden="true"
          class="z-10 opacity-0 group-data-[selected=true]:opacity-100 w-4 h-3 transition-opacity motion-reduce:transition-none"
          role="presentation"
          viewBox="0 0 17 18"
        >
          <polyline
            fill="none"
            points="1 9 7 14 15 4"
            stroke="currentColor"
            stroke-dasharray="22"
            stroke-dashoffset="44"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            style="transition: stroke-dashoffset 250ms linear 0.2s;"
          />
        </svg>
      </span>
      <span
        class="relative text-foreground select-none text-medium transition-colors-opacity before:transition-width motion-reduce:transition-none"
        id=":ra:"
      >
        <p
          class="text-primary dark:text-white text-medium leading-xl"
        >
          Keep me logged in
        </p>
      </span>
    </label>
    <a
      class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
      href="/forget-password"
      role="link"
      tabindex="0"
    >
      <p
        class="text-blue-450 dark:text-white text-small font-medium pr-3"
      >
        Forget password?
      </p>
    </a>
  </div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover text-black dark:text-white opacity-50 cursor-not-allowed bg-blue-450 dark:bg-purple-750 w-full py-7 mb-6 mt-8"
    data-disabled="true"
    disabled=""
    type="button"
  >
    <p
      class="text-primary dark:text-white text-small text-white font-bold leading-4"
    >
      Sign In
    </p>
  </button>
</div>
`;
